File: mobile_apps/ALADDIN_iOS/ALADDIN/Core/Models/APIModels.swift
import Foundation
import SwiftUI

/**
 * 📦 API Models
 * Модели данных для API запросов и ответов
 * Соответствуют структуре вашего Python backend
 */

// MARK: - VPN Models

struct VPNStatusResponse: Codable {
    let isConnected: Bool
    let serverLocation: String
    let ipAddress: String
    let ping: Int
    let downloadSpeed: String
    let uploadSpeed: String
    let sessionTime: String
    let threatsBlocked: Int
}

struct VPNServer: Codable, Identifiable {
    let id: String
    let country: String
    let city: String
    let flag: String
    let ping: Int
    let load: Int // 0-100%
}

// MARK: - Family Models

struct FamilyMemberResponse: Codable, Identifiable {
    let id: String
    let name: String
    let role: String // "parent", "child", "teenager", "elderly"
    let avatar: String
    let status: String // "protected", "warning", "danger", "offline"
    let threatsBlocked: Int
    let lastActive: String
    let devices: Int
}

struct FamilyStatsResponse: Codable {
    let totalMembers: Int
    let totalDevices: Int
    let totalThreats: Int
    let protectionLevel: Int
}

// MARK: - Analytics Models

struct AnalyticsResponse: Codable {
    let period: String // "day", "week", "month"
    let threatsDetected: Int
    let threatsBlocked: Int
    let itemsScanned: Int
    let protectionLevel: Int
    let topThreats: [ThreatItem]
    let threatsByType: [ThreatByType]
}

struct ThreatItem: Codable, Identifiable {
    let id: String
    let name: String
    let count: Int
    let icon: String
    let severity: String // "low", "medium", "high", "critical"
}

struct ThreatByType: Codable {
    let type: String // "web", "app", "network", "file"
    let count: Int
    let percentage: Double
}

// MARK: - AI Assistant Models

struct ChatMessageRequest: Codable {
    let message: String
    let userId: String
    let timestamp: Date
}

struct ChatMessageResponse: Codable {
    let message: String
    let timestamp: Date
    let suggestions: [String]?
}

// MARK: - User Models

struct UserProfile: Codable {
    let id: String
    let name: String
    let email: String
    let phone: String?
    let registrationDate: String
    let subscriptionType: String
    let subscriptionEndDate: String?
    let threatsBlocked: Int
    let familyMembers: Int
    let devices: Int
}

struct UpdateProfileRequest: Codable {
    let name: String?
    let email: String?
    let phone: String?
}

// MARK: - Parental Control Models

struct ParentalControlSettings: Codable {
    let childId: String
    let isContentFilterEnabled: Bool
    let isAppBlockingEnabled: Bool
    let screenTimeLimitHours: Int
    let allowedApps: [String]
    let blockedWebsites: [String]
    let bedtime: String? // "22:00"
}

struct ChildStatsResponse: Codable {
    let screenTimeToday: String
    let blockedSitesToday: Int
    let allowedAppsCount: Int
    let timeRemaining: String
}

// MARK: - Notifications Models

struct NotificationResponse: Codable, Identifiable {
    let id: String
    let icon: String
    let title: String
    let message: String
    let timestamp: Date
    let isRead: Bool
    let type: String // "threat", "success", "info", "warning"
}

// MARK: - Subscription Models

struct TariffResponse: Codable, Identifiable {
    let id: String
    let name: String
    let price: Int
    let period: String
    let features: [String]
    let isRecommended: Bool
}

struct SubscriptionStatus: Codable {
    let isActive: Bool
    let tariffId: String
    let startDate: Date
    let endDate: Date
    let autoRenew: Bool
}

// MARK: - Auth Models

struct LoginRequest: Codable {
    let email: String
    let password: String
}

struct LoginResponse: Codable {
    let token: String
    let userId: String
    let expiresAt: Date
}

struct RegisterRequest: Codable {
    let name: String
    let email: String
    let password: String
    let phone: String?
}

// MARK: - Generic Response

struct APIResponse<T: Codable>: Codable {
    let success: Bool
    let data: T?
    let message: String?
    let error: String?
}

File: mobile_apps/ALADDIN_Android/app/src/main/java/family/aladdin/android/models/ApiModels.kt
package family.aladdin.android.models

import com.google.gson.annotations.SerializedName
import java.util.Date

/**
 * 📦 API Models
 * Модели данных для API запросов и ответов (Android)
 * Соответствуют структуре Python backend
 */

// ═══════════════════════════════════════════════════════════════
// Payment Models (QR оплата)
// ═══════════════════════════════════════════════════════════════

data class MerchantInfo(
    val name: String,
    val card: String,
    val phone: String
)

// ═══════════════════════════════════════════════════════════════
// VPN Models
// ═══════════════════════════════════════════════════════════════

data class VPNStatusResponse(
    val isConnected: Boolean,
    val serverLocation: String,
    val ipAddress: String,
    val ping: Int,
    val downloadSpeed: String,
    val uploadSpeed: String,
    val sessionTime: String,
    val threatsBlocked: Int
)

data class VPNServer(
    val id: String,
    val country: String,
    val city: String,
    val flag: String,
    val ping: Int,
    val load: Int // 0-100%
)

// ═══════════════════════════════════════════════════════════════
// Family Models
// ═══════════════════════════════════════════════════════════════

data class FamilyMemberResponse(
    val id: String,
    val name: String,
    val role: String, // "parent", "child", "teenager", "elderly"
    val avatar: String,
    val status: String, // "protected", "warning", "danger", "offline"
    val threatsBlocked: Int,
    val lastActive: String,
    val devices: Int
)

data class FamilyStatsResponse(
    val totalMembers: Int,
    val totalDevices: Int,
    val totalThreats: Int,
    val protectionLevel: Int
)

// ═══════════════════════════════════════════════════════════════
// Analytics Models
// ═══════════════════════════════════════════════════════════════

data class AnalyticsResponse(
    val period: String, // "day", "week", "month"
    val threatsDetected: Int,
    val threatsBlocked: Int,
    val itemsScanned: Int,
    val protectionLevel: Int,
    val topThreats: List<ThreatItem>,
    val threatsByType: List<ThreatByType>
)

data class ThreatItem(
    val id: String,
    val name: String,
    val count: Int,
    val icon: String,
    val severity: String // "low", "medium", "high", "critical"
)

data class ThreatByType(
    val type: String, // "web", "app", "network", "file"
    val count: Int,
    val percentage: Double
)

// ═══════════════════════════════════════════════════════════════
// AI Assistant Models
// ═══════════════════════════════════════════════════════════════

data class ChatMessageRequest(
    val message: String,
    val userId: String,
    val timestamp: Long = System.currentTimeMillis()
)

data class ChatMessageResponse(
    val message: String,
    val timestamp: Long,
    val suggestions: List<String>?
)

// ═══════════════════════════════════════════════════════════════
// User Models
// ═══════════════════════════════════════════════════════════════

data class UserProfile(
    val id: String,
    val name: String,
    val email: String,
    val phone: String?,
    val registrationDate: String,
    val subscriptionType: String,
    val subscriptionEndDate: String?,
    val threatsBlocked: Int,
    val familyMembers: Int,
    val devices: Int
)

data class UpdateProfileRequest(
    val name: String?,
    val email: String?,
    val phone: String?
)

// ═══════════════════════════════════════════════════════════════
// Parental Control Models
// ═══════════════════════════════════════════════════════════════

data class ParentalControlSettings(
    val childId: String,
    val isContentFilterEnabled: Boolean,
    val isAppBlockingEnabled: Boolean,
    val screenTimeLimitHours: Int,
    val allowedApps: List<String>,
    val blockedWebsites: List<String>,
    val bedtime: String? // "22:00"
)

data class ChildStatsResponse(
    val screenTimeToday: String,
    val blockedSitesToday: Int,
    val allowedAppsCount: Int,
    val timeRemaining: String
)

// ═══════════════════════════════════════════════════════════════
// Notifications Models
// ═══════════════════════════════════════════════════════════════

data class NotificationResponse(
    val id: String,
    val icon: String,
    val title: String,
    val message: String,
    val timestamp: Long,
    val isRead: Boolean,
    val type: String // "threat", "success", "info", "warning"
)

// ═══════════════════════════════════════════════════════════════
// Subscription Models
// ═══════════════════════════════════════════════════════════════

data class TariffResponse(
    val id: String,
    val name: String,
    val price: Int,
    val period: String,
    val features: List<String>,
    val isRecommended: Boolean
)

data class SubscriptionStatus(
    val isActive: Boolean,
    val tariffId: String,
    val startDate: Long,
    val endDate: Long,
    val autoRenew: Boolean
)

// ═══════════════════════════════════════════════════════════════
// Auth Models
// ═══════════════════════════════════════════════════════════════

data class LoginRequest(
    val email: String,
    val password: String
)

data class LoginResponse(
    val token: String,
    val userId: String,
    val expiresAt: Long
)

data class RegisterRequest(
    val name: String,
    val email: String,
    val password: String,
    val phone: String?
)

// ═══════════════════════════════════════════════════════════════
// Generic Response
// ═══════════════════════════════════════════════════════════════

data class ApiResponse<T>(
    val success: Boolean,
    val data: T?,
    val message: String?,
    val error: String?
)

